pkg/plugin/builtin/v1/internal/fromplugin/specifier.go

```diff
@@ -40,15 +58,28 @@ func SpecifierSpecifyResponse(in cpluginv1.SpecifierSpecifyResponse) (plugin.Spe
 }
 
 func SpecifierParameter(in cpluginv1.SpecifierParameter) plugin.Parameter {
-	var validations []plugin.Validation
-	if in.Required {
+	validations := make([]plugin.Validation, len(in.Validations))
+
+	requiredExists := false
+	for i, v := range in.Validations {
+		validations[i] = plugin.Validation{
+			Type:  plugin.ValidationType(v.Type),
+			Value: v.Value,
+		}
+		if v.Type == cpluginv1.ValidationTypeRequired {
+			requiredExists = true
+		}
+	}
+	// making sure not to duplicate the required validation
+	if in.Required && !requiredExists {
 		validations = append(validations, plugin.Validation{
 			Type: plugin.ValidationTypeRequired,
 		})
 	}
+
 	return plugin.Parameter{
 		Default:     in.Default,
-		Type:        "string",
+		Type:        plugin.ParameterType(in.Type),
```

Comment:
We should do the same transformation here for unspecified parameters as we do for standalone plugins, just so we ensure the behavior is the same.